service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala

```diff
@@ -203,16 +203,31 @@ case class DefaultMahaCalciteSqlParser(mahaServiceConfig: MahaServiceConfig) ext
   }
 
   def getFilterList(sqlNode: SqlNode, publicFact: PublicFact) : (IndexedSeq[Filter], Option[Filter], Option[Filter], Option[Filter], Int) = {
-    sqlNode match {
+    val sqlWhere = sqlNode.asInstanceOf[SqlSelect].getWhere
+    val sqlHaving = sqlNode.asInstanceOf[SqlSelect].getHaving
+    var filterList = ArrayBuffer.empty[Filter]
+
+    sqlWhere match {
       case sqlBasicCall: SqlBasicCall =>
-        val filterList = constructFilters(sqlBasicCall)
-        validate(filterList)
+        filterList = filterList ++ constructFilters(sqlBasicCall)
       case others: AnyRef =>
         logger.error(s"sqlNode type ${sqlNode.getKind} in getSelectList is not yet supported")
         (IndexedSeq.empty, None, None, None, 1)
       case null =>
-        logger.error("sqlNode is null in getFilterList")
+        logger.error("Empty WHERE clause.")
+        (IndexedSeq.empty, None, None, None, 1)
+    }
+
+    sqlHaving match {
+      case sqlBasicCall: SqlBasicCall =>
+        filterList = filterList ++ constructFilters(sqlBasicCall)
+        validate(filterList)
+      case others: AnyRef =>
+        logger.error(s"sqlNode type ${sqlNode.getKind} in getSelectList is not yet supported")
         (IndexedSeq.empty, None, None, None, 1)
+      case null =>
+        logger.debug("Empty HAVING clause.")
+        validate(filterList)
```

Comment:
for null having, we dont need to validate filters. We should pull validate filters outside of where and having and invoke it always.

Reply:
Fixed